### PR TITLE
Collection modal interaction

### PR DESCRIPTION
**Pull request by rwd**

### CODE REVIEW

File: components/account/ModalCollection.vue [hunk 1]
```diff
@@ -126,13 +185,14 @@
 <style lang="scss" scoped>
   .btn-collection {
```
Comment (mirjamverloop) [inline components/account/ModalCollection.vue:186]: When the button has a background image, the text is kind of hard to read. Suggestion: 

```
diff --git a/components/account/ModalCollection.vue b/components/account/ModalCollection.vue
index ecde5157..3914f637 100644
--- a/components/account/ModalCollection.vue
+++ b/components/account/ModalCollection.vue
@@ -77,7 +77,7 @@
           class="btn-collection w-100 text-left"
           @click="addItem(collection.id)"
         >
-          {{ displayField(collection, 'title') }} ({{ collection.visibility }}) - {{ collection.total }} items
+          <span>{{ displayField(collection, 'title') }} ({{ collection.visibility }}) - {{ collection.total }} items</span>
         </b-button>
       </div>
       <div class="modal-footer">
@@ -188,7 +188,25 @@
     font-weight: 500;
     margin-bottom: 0.5rem;
     padding: 1rem;
+    position: relative;
     text-transform: none;
+
+    &.btn-overlay {
+      span {
+        position: relative;
+        z-index: 10;
+      }
+
+      &:after {
+        content: '';
+        background: rgba(0, 0, 0, 0.7);
+        height: 100%;
+        position: absolute;
+        top: 0;
+        left: 0;
+        width: 100%;
+      }
+    }
   }
 
   .collections {
```

File: components/account/ModalCollection.vue [hunk 2]
```diff
@@ -1,75 +1,91 @@
...
+    :id="modalId"
     hide-header
     hide-footer
+    @show="fetchCollections"
   >
     <div v-if="showForm">
-      <h2>Create new collection</h2>
+      <h1 class="pt-2">
+        {{ $t('collectionModal.createNewCollection') }}
+      </h1>
```
[the remark this answers is not in the document]
Comment (rwd) [reply components/account/ModalCollection.vue]: I have reverted to using the title attribute with the Bootstrap Modal default element and styling.

File: components/item/ItemPreviewCardGroup.vue [hunk 3]
```diff
@@ -19,11 +19,23 @@
...
+      id="new-collection-toast"
+      toast-class="brand-toast"
+      toaster="b-toaster-bottom-left"
+      auto-hide-delay="5000"
+      is-status
+      no-close-button
+      solid
+      data-qa="tier toast"
+    >
+      {{ $t('collectionModal.newNotification') }}
```
[the remark this answers is not in the document]
Comment (rwd) [reply components/item/ItemPreviewCardGroup.vue:32]: Not sure. I have removed that.

File: components/account/ModalCollection.vue [hunk 4]
```diff
@@ -1,75 +1,91 @@
...
-          label="Collection name"
+          :label="$t('collectionModal.collectionName')"
           label-for="collection"
         >
           <b-form-input
             id="collection"
+            v-model="newCollectionName"
             type="text"
             maxlength="35"
             required
```
[the remark this answers is not in the document]
Comment (rwd) [reply components/account/ModalCollection.vue]: I think that should be addressed later on.

File: components/account/ModalCollection.vue (hunk 1 above)
Comment (rwd) [reply components/account/ModalCollection.vue]: Agreed, and applied.

File: components/account/CollectionModal.vue [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+      submitForm() {
+        // TODO: Create set-plugin call for creating a new set
+        // TODO: Add this item (its id can be found in the store) to the new set
+
+        // TODO: should this emit an event and allow the parent component to toast?
+        this.$bvToast.show('new-collection-toast');
+        this.newCollectionName = '';
+        this.newCollectionDescription = '';
+        this.newCollectionPrivate = false;
```
Comment (mirjamverloop) [inline components/account/CollectionModal.vue:158]: This seems a duplication, as the next call is goBack, which also sets these to empty or false.

File: components/account/CollectionModal.vue [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+      };
+    },
+
+    mounted() {
+      this.$root.$on('bv::modal::hidden', () => {
+        this.showForm = false;
+      });
+    },
+
+    computed: {
```
Comment (mirjamverloop) [inline components/account/CollectionModal.vue:121]: The computed property should go before the mounted.

File: components/account/ModalCollection.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mirjamverloop) [reply components/account/ModalCollection.vue]: Agreed!

**Pull request merged**